### PR TITLE
Add InjectableField decorator

### PR DESCRIPTION
**Pull request by oliverguenther**
Adds `@InjectField` to make it easier to inject fields outside the constructor:

```
@InjectField() myService:Service
```

instead of 

```
myService:Service = this.injector.get(Service);
```

Works on all classes with a public injector. Will raise a TS error if injector is not accessible

### CODE REVIEW

File: frontend/src/app/components/work-packages/work-package-comment/wp-comment-field.component.ts [hunk 1]
```diff
@@ -32,14 +32,15 @@ import {
   FormattableEditFieldComponent,
   formattableFieldTemplate
 } from "core-app/modules/fields/edit/field-types/formattable-edit-field.component";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/components/work-packages/work-package-comment/wp-comment-field.component.ts:36]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/components/work-packages/work-package-comment/wp-comment-field.component.ts#L36"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-table/embedded/wp-embedded-table.component.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {QueryFormResource} from "core-app/modules/hal/resources/query-form-resou
 import {QueryFormDmService} from "core-app/modules/hal/dm-services/query-form-dm.service";
 import {distinctUntilChanged, take, withLatestFrom, map} from "rxjs/operators";
 import {untilComponentDestroyed} from "ng2-rx-componentdestroyed";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-table/embedded/wp-embedded-table.component.ts:16]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/components/wp-table/embedded/wp-embedded-table.component.ts#L16"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts [hunk 3]
```diff
@@ -64,15 +64,13 @@ import {
   withLoadingIndicator
 } from "core-app/modules/common/loading-indicator/loading-indicator.service";
 import {combineLatest} from "rxjs";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts:68]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts#L68"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts [hunk 4]
```diff
@@ -30,6 +30,7 @@ import {Component} from "@angular/core";
 import * as moment from "moment";
 import {TimezoneService} from "core-components/datetime/timezone.service";
 import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.component";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts:34]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/date-edit-field.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/duration-edit-field.component.ts [hunk 5]
```diff
@@ -30,6 +30,7 @@ import {TimezoneService} from 'core-components/datetime/timezone.service';
 import * as moment from 'moment';
 import {Component} from "@angular/core";
 import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.component";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/duration-edit-field.component.ts:34]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/duration-edit-field.component.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts [hunk 6]
```diff
@@ -34,14 +34,15 @@ import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.compon
 import {ValueOption} from "core-app/modules/fields/edit/field-types/select-edit-field.component";
 import {NgSelectComponent} from "@ng-select/ng-select";
 import {untilComponentDestroyed} from "ng2-rx-componentdestroyed";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts:38]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L38"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/project-status-edit-field.component.ts [hunk 7]
```diff
@@ -31,15 +31,15 @@ import {Component, OnInit, ViewChild} from "@angular/core";
 import {EditFieldComponent} from "core-app/modules/fields/edit/edit-field.component";
 import {NgSelectComponent} from "@ng-select/ng-select";
 import {projectStatusCodeCssClass, projectStatusI18n} from "core-app/modules/fields/helpers/project-status-helper";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/project-status-edit-field.component.ts:35]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/project-status-edit-field.component.ts#L35"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts [hunk 8]
```diff
@@ -35,14 +35,16 @@ import {
   TimeEntryWorkPackageAutocompleterMode
 } from "core-app/modules/common/autocomplete/te-work-package-autocompleter.component";
 import {HalResource} from "core-app/modules/hal/resources/hal-resource";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
 
 const RECENT_TIME_ENTRIES_MAGIC_NUMBER = 30;
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts:41]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts#L41"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/global_search/title/global-search-title.component.ts [hunk 9]
```diff
@@ -40,6 +40,7 @@ import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
 import {GlobalSearchService} from "core-app/modules/global_search/services/global-search.service";
 import {CurrentProjectService} from "core-components/projects/current-project.service";
 import {Injector} from "@angular/core";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
 
 export const globalSearchTitleSelector = 'global-search-title';
```
Comment (openprojectci) [inline frontend/src/app/modules/global_search/title/global-search-title.component.ts:46]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/global_search/title/global-search-title.component.ts#L46"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts [hunk 10]
```diff
@@ -38,6 +38,7 @@ import {KeepTabService} from 'core-components/wp-single-view-tabs/keep-tab/keep-
 import {FirstRouteService} from "core-app/modules/router/first-route-service";
 import {WorkPackageSingleViewBase} from "core-app/modules/work_packages/routing/wp-view-base/work-package-single-view.base";
 import {BackRoutingService} from "core-app/modules/common/back-routing/back-routing.service";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
```
Comment (openprojectci) [inline frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts:42]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/work_packages/routing/wp-full-view/wp-full-view.component.ts#L42"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/components/wp-table/timeline/global-elements/wp-timeline-relations.directive.ts#L79">frontend/src/app/components/wp-table/timeline/global-elements/wp-timeline-relations.directive.ts#L79</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46">frontend/src/app/modules/common/editable-toolbar-title/editable-toolbar-title.component.ts#L46</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48">frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/c7c24daa011e49abe1a4ee0f050d60d8974f618f/frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts#L46">frontend/src/app/modules/fields/edit/field-types/select-edit-field.component.ts#L46</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: frontend/src/app/components/main-menu/main-menu-toggle.service.ts [hunk 11]
```diff
@@ -40,7 +41,8 @@ export class MainMenuToggleService {
   private elementWidth:number;
   private readonly localStorageKey:string = 'openProject-mainMenuWidth';
   private readonly defaultWidth:number = 230;
-  private readonly currentProject:CurrentProjectService = this.injector.get(CurrentProjectService);
```
Comment (HDinger) [inline frontend/src/app/components/main-menu/main-menu-toggle.service.ts:43]: For whatever reason this was again called in line [79](https://github.com/opf/openproject/blob/0d5d295fec5e57b4542a818c3f681c2be34b856b/frontend/src/app/components/main-menu/main-menu-toggle.service.ts#L79). I guess the second one can be removed, too.

**Review by HDinger — APPROVED** [review]
Good job 👍 I couldn't catch any issues. However I found some places where the `injector.get` syntax is still used (e.g. UserFieldPortalComponent, DragDropHandleBuilder, WpTableConfigurationModalComponent, OpenProjectPluginContext). Is there a reason, these were not replaced?

**Pull request merged**